[4.5] docs(dropdownbutton): Documentation for DropDownButton (#1595)
telerik/blazor-docs: components/dropdownbutton/icons.md

---
title: Icons

> page_title: DropDownButton Icons

Reviewer: This seems like a pattern used by other feature-related articles.
```suggestion
page_title: Blazor DropDownButton - Icons
```

Author: That's correct, there is such a patter. However, we do not include the Blazor keyword in the page title. I will update the titles of all articles.

> description: How to use icons with the DropDownButton for Blazor.

Reviewer: See the comment in `overview.md`.

slug: dropdownbutton-icons
tags: telerik,blazor,dropdownbutton,icons
published: True
position: 5
---

# DropDownButton Icons

> The `<TelerikDropDownButton>` and `<DropDownButtonItem>` provide an `Icon` parameter to add an icon to the main button and all secondary action items.

Reviewer: ```suggestion
The `<TelerikDropDownButton>` and `<DropDownButtonItem>` provide an `Icon` parameter that allows you to add an icon to the main button and all secondary action items.
```

The `Icon` parameter type is `object` and it accepts:

* a member of the [`FontIcon` enum]({%slug general-information/font-icons%}#icons-list)
* a property of the static [`SvgIcon` class]({%slug general-information/font-icons%}#icons-list)
* a `string` that is a CSS class for a custom icon

>caption How to use icons in Telerik Blazor DropDownButton

````CSHTML
<TelerikDropDownButton Icon="@("sln")">
    <DropDownButtonContent>Telerik Icon</DropDownButtonContent>
    <DropDownButtonItems>
        <DropDownButtonItem Icon="@FontIcon.User">Telerik Font Icon</DropDownButtonItem>
        <DropDownButtonItem Icon="@SvgIcon.Gear">Telerik SVG Icon</DropDownButtonItem>
        <DropDownButtonItem Icon="@CustomIconClass">Custom Icon</DropDownButtonItem>
        <DropDownButtonItem> <TelerikLoader /> Custom markup </DropDownButtonItem>
    </DropDownButtonItems>
</TelerikDropDownButton>

<style>
    /* Third-party icon libraries should provide these styles out-of-the-box. */

    .my-icon {
        width: 1em;
        height: 1em;
        font-size: 16px;
        background: purple;
    }
</style>

@code {
    string CustomIconClass { get; set; } = "my-icon";
}
````

> >tip You can also add custom icons and images with additional markup inside the `<DropDownButtonContent>`, where the text is.

Reviewer: The statement is somewhat ambiguous - "with" in "custom icons and images with additional markup" can also mean "through". If this is the case, you must phrase it.

> >note Images used as icons should generally be small enough to fit in a line of text. The primary button is an inline element and is not designed for large images. If you want to use big icon buttons, consider one of the following options:

Reviewer: This is a huge note that looks funny once you build the article. Consider turning it into a new document section by adding an H2, for example, "## Best Practices".

![image](https://github.com/telerik/blazor-docs/assets/60105689/3435df3a-f86e-4401-9220-7c32faee2bc4)

>

> > * Define a `Class` on the button that provides `height` and `width`. The width and height can be set in `px` sufficient to accommodate the icon or to `auto`,

Reviewer: ```suggestion
> * Define a `Class` for the button that provides `height` and `width`. The width and height can be set to `auto` or to a value in `px`. When using a static value, it must be sufficient to accommodate the icon.
```
